File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Client/src/models/impersonationContext/impersonationContext.ts
export class ImpersonationContext {
    public id: string = "";
    public requesterEntraUserId: string = "";
    public requesterFirstName: string = "";
    public requesterLastName: string = "";
    public requesterDisplayName: string = "";
    public requesterEmail: string = "";
    public requesterJobTitle: string = "";
    public responsiblePersonEntraUserId: string = "";
    public responsiblePersonFirstName: string = "";
    public responsiblePersonLastName: string = "";
    public responsiblePersonDisplayName: string = "";
    public responsiblePersonEmail: string = "";
    public responsiblePersonJobTitle: string = "";
    public reason: string = "";
    public purpose?: string = "";
    public organisation: string = "";
    public projectName: string = "";
    public inboxSasToken?: string = "";
    public outboxSasToken?: string = "";
    public errorsSasToken?: string = "";
    public isApproved: boolean = false;
    public identifierColumn: string = "";
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;
}

File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server.Tests.Acceptance/Models/ImpersonationContexts/ImpersonationContext.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Portals.Server.Tests.Acceptance.Models.ImpersonationContexts
{
    public class ImpersonationContext
    {
        public Guid Id { get; set; }
        public string RequesterEntraUserId { get; set; } = string.Empty;
        public string RequesterFirstName { get; set; } = string.Empty;
        public string RequesterLastName { get; set; } = string.Empty;
        public string RequesterDisplayName { get; set; } = string.Empty;
        public string RequesterEmail { get; set; } = string.Empty;
        public string RequesterJobTitle { get; set; } = string.Empty;
        public string ResponsiblePersonEntraUserId { get; set; } = string.Empty;
        public string ResponsiblePersonFirstName { get; set; } = string.Empty;
        public string ResponsiblePersonLastName { get; set; } = string.Empty;
        public string ResponsiblePersonDisplayName { get; set; } = string.Empty;
        public string ResponsiblePersonEmail { get; set; } = string.Empty;
        public string ResponsiblePersonJobTitle { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public string Purpose { get; set; } = string.Empty;
        public string Organisation { get; set; } = string.Empty;
        public string ProjectName { get; set; } = string.Empty;
        public string InboxSasToken { get; set; } = string.Empty;
        public string OutboxSasToken { get; set; } = string.Empty;
        public string ErrorsSasToken { get; set; } = string.Empty;
        public bool IsApproved { get; set; }
        public string IdentifierColumn { get; set; } = string.Empty;
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
    }
}
